test/rules/object-literal-key-quotes/always/test.ts.fix

```diff
@@ -6,8 +6,8 @@ const o = {
   "hyphens-need-quotes": null,
   [computed]: 456,
   "123": "hello", // failure
-  "1e4": "something", // failure
-  ".123": "float", // failure
+  "10000": "something", // failure
```

Comment:
`1e4` is a numeric literal that is automatically expanded to 1000 by the javascript vm before stringifying. Fixing this to `"1e4"` would actually break the program.

```ts
> Object.keys({1e4: "foo"})
[ '10000' ]
> String(1e4)==="10000"
true
```